### ISSUE TITLE
[java] AccessorClassGeneration false-negative: subclass calls private constructor

### ISSUE THREAD

**Issue by Vampire** [pmd/pmd#1998]
Rule: [AccessorClassGeneration](https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#accessorclassgeneration)

Possibly a false-negative.

Having a sub-class that calls a private constructor is bad practice.
It requires the compiler adding another package-private synthetic constructor with an additional synthetic anonymous inner class parameter that forwards to the private constructor so that the subclass has something non-private to call.

Examples for this would be for example

```java
class Foo {
    private Foo() { }
    public Foo(String foo) { }
    private static class Bar extends Foo { }
}
```

where the default constructor of Bar calls the private no-arg constructor of Foo or

```java
class Foo {
    private Foo() { }
    private Foo(String foo) { }
    public Foo(String foo, String bar) { }
    private static class Bar extends Foo {
        public Bar() { super(null); }
    }
}
```

where the call to the private constructor is explicit.

This adds unnecessarily additional constructors and also can cause problems if you use the class from Groovy code, as in the first example `new Foo(null)` and in the second example `new Foo(null, null)` can not be executed as there is a disambiguity between the constructor you meant to call and the synthetic added constructor.

It is usually better practice to simply make that constructor package-private direclty, eliminating the need for any synthetic constructors.

**Comment by jsotuyod** [comment]
@Vampire this should be already covered by [`java-bestpractices/AccessorClassGeneration`](https://pmd.github.io/pmd-6.17.0/pmd_rules_java_bestpractices.html#accessorclassgeneration).

It's possible however that the rule misses these scenarios, as there is no explicit call to the private constructor / it's through a call to `super`, in which case this should be flagged as a false negative. Could you please confirm this and update the report accordingly?

**Comment by adangel** [comment]
Fixed via #2899 for PMD 7.
